apps/api/src/app/inbox/usecases/snooze-notification/snooze-notification.usecase.ts

```diff
@@ -0,0 +1,242 @@
+  NotImplementedException,
+  HttpStatus,
+} from '@nestjs/common';
+import {
+  CreateExecutionDetails,
+  CreateExecutionDetailsCommand,
+  DetailEnum,
+  StandardQueueService,
+  FeatureFlagsService,
+  SYSTEM_LIMITS,
+  PinoLogger,
+} from '@novu/application-generic';
+import {
+  JobEntity,
+  JobRepository,
+  MessageRepository,
+  MessageEntity,
+  OrganizationEntity,
+  EnvironmentEntity,
+  UserEntity,
+  CommunityOrganizationRepository,
+} from '@novu/dal';
+import {
+  ApiServiceLevelEnum,
+  ChannelTypeEnum,
+  ExecutionDetailsSourceEnum,
+  ExecutionDetailsStatusEnum,
+  FeatureFlagsKeysEnum,
+  FeatureNameEnum,
+  getFeatureForTierAsNumber,
+  JobStatusEnum,
+} from '@novu/shared';
+import { v4 as uuidv4 } from 'uuid';
+import { SnoozeNotificationCommand } from './snooze-notification.command';
+import { MarkNotificationAs } from '../mark-notification-as/mark-notification-as.usecase';
+import { MarkNotificationAsCommand } from '../mark-notification-as/mark-notification-as.command';
+import { InboxNotification } from '../../utils/types';
+
+const LOG_CONTEXT = 'SnoozeNotification';
+
+@Injectable()
+export class SnoozeNotification {
+  private readonly RETRY_ATTEMPTS = 3;
+
+  constructor(
+    private readonly logger: PinoLogger,
+    private messageRepository: MessageRepository,
+    private jobRepository: JobRepository,
+    private standardQueueService: StandardQueueService,
+    private organizationRepository: CommunityOrganizationRepository,
+    private createExecutionDetails: CreateExecutionDetails,
+    private markNotificationAs: MarkNotificationAs,
+    private featureFlagsService: FeatureFlagsService
+  ) {}
+
+  public async execute(command: SnoozeNotificationCommand): Promise<InboxNotification> {
+    await this.isSnoozeEnabled(command);
+
+    const notification = await this.findNotification(command);
+    const delayAmount = this.calculateDelayInMs(command.snoozeUntil);
+    await this.validateDelayDuration(command, delayAmount);
+
+    try {
+      let scheduledJob = {} as JobEntity;
+      let snoozedNotification = {} as InboxNotification;
+
+      await this.messageRepository.withTransaction(async () => {
+        scheduledJob = await this.createScheduledUnsnoozeJob(notification, delayAmount);
+        snoozedNotification = await this.markNotificationAsSnoozed(command);
+        await this.enqueueJob(scheduledJob, delayAmount);
+      });
+
+      // fire and forget
+      this.createExecutionDetails
+        .execute(
+          CreateExecutionDetailsCommand.create({
+            ...CreateExecutionDetailsCommand.getDetailsFromJob(scheduledJob),
+            detail: DetailEnum.MESSAGE_SNOOZED,
+            source: ExecutionDetailsSourceEnum.INTERNAL,
+            status: ExecutionDetailsStatusEnum.PENDING,
+            isTest: false,
+            isRetry: false,
+          })
+        )
+        .catch((error) => {
+          this.logger.error({ err: error }, 'Failed to create execution details', LOG_CONTEXT);
```

Comment:
unfortunately, this won’t fully print the context. 😞
what I actually wanted was to create a new interface for the logger to make it stricter and safer to use.
however, without rebuilding the entire LoggerModule, I don't see a clean way to enforce stricter typing or structure around the logger itself.

so at the moment, I’m limited to what we have.

To solve it you would need to `this.logger.setContext(this.constructor.name);`, then remove the context in this line as it redundant and wont be used.
```suggestion
          this.logger.error({ err: error }, 'Failed to create execution details');
```